internal/grpc/services/gateway/gateway.go

```diff
@@ -69,6 +69,7 @@ type config struct {
 	HomeMapping         string                            `mapstructure:"home_mapping"`
 	TokenManagers       map[string]map[string]interface{} `mapstructure:"token_managers"`
 	EtagCacheTTL        int                               `mapstructure:"etag_cache_ttl"`
+	HiddenRootFolders   map[string][]string               `mapstructure:"hidden_root_folders"`
```

Comment:
Have it the other way round, `path -> userAgentList`. Call it `AllowedUserAgents`